edn/src/types.rs

```diff
@@ -0,0 +1,73 @@
+
+use std::collections::{BTreeSet, BTreeMap, LinkedList};
+use std::cmp::{Ordering, Ord, PartialOrd};
+use ordered_float::OrderedFloat;
+
+/// We're using BTree{Set, Map} rather than Hash{Set, Map} because the BTree variants implement Hash
+/// (unlike the Hash variants which don't in order to preserve O(n) hashing time which is hard given
+/// recurrsive data structures)
+/// See https://internals.rust-lang.org/t/implementing-hash-for-hashset-hashmap/3817/1
+/// TODO: We should probably Box the collection types
```

Comment:
Well TBH I'm copying @jimblandy who uses `Object(Box<HashMap<String, Json>>)` in his JSON example (see enum chapter). Except that I note [the reality](https://github.com/serde-rs/json/blob/a32de277b50cbbbfd786d0a93f566dd30d71a537/json/src/value.rs#L97) is different.